packages/kbn-management/settings/application/services.tsx

```diff
@@ -6,17 +6,37 @@
  * Side Public License, v 1.
  */
 
-import React, { FC } from 'react';
+import React, { FC, useContext } from 'react';
 
 import {
   FormProvider,
   FormKibanaProvider,
   type FormKibanaDependencies,
   type FormServices,
 } from '@kbn/management-settings-components-form';
+import { UiSettingMetadata } from '@kbn/management-settings-types';
+import { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+import { normalizeSettings } from '@kbn/management-settings-utilities';
+import { Subscription } from 'rxjs';
 
-export type SettingsApplicationServices = FormServices;
-export type SettingsApplicationKibanaDependencies = FormKibanaDependencies;
+export interface Services {
+  getAllowlistedSettings: () => Record<string, UiSettingMetadata>;
+  subscribeToUpdates: (fn: () => void) => Subscription;
+  isCustomSetting: (key: string) => boolean;
+  isOverriddenSetting: (key: string) => boolean;
+}
+
+export type SettingsApplicationServices = Services & FormServices;
+
+export interface KibanaDependencies {
+  settings: {
+    client: Pick<IUiSettingsClient, 'getAll' | 'isCustom' | 'isOverridden' | 'getUpdate$'>;
```

Comment:
We can `Pick` only the methods we use in this component/its services, making it easier to not only mock, but to understand (and track).